applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -483,6 +499,30 @@ void DistanceModificationProcess::SetDiscontinuousDistanceToSplitFlag()
     }
 }
 
+void DistanceModificationProcess::CheckAndStoreVariablesList(const std::vector<std::string>& rVariableStringArray) 
+{
+    const auto& r_node = *mrModelPart.NodesBegin(); 
+    if (rVariableStringArray.size()>0){
+        for (std::size_t i_variable=0; i_variable < rVariableStringArray.size(); i_variable++){
+            if (KratosComponents<Variable<double>>::Has(rVariableStringArray[i_variable])) {
+                KRATOS_CHECK_DOF_IN_NODE(KratosComponents<Variable<double>>::Get(rVariableStringArray[i_variable]),r_node);
+               
+                const Variable<double>* p_double_var  = &KratosComponents<Variable<double>>::Get(rVariableStringArray[i_variable]);
```

Comment:
Very minor but, if you do this before the check, you can avoid getting the variable twice. Also, since you are checking (:+1:, good idea), you could also add `KRATOS_CHECK_VARIABLE_IN_NODAL_DATA`

Reply:
Also, I know I suggested the pointer, but using the value would be much cleaner. The only thing keeping me from recommending that is that we have had some recent issues with copying variables, but feel free to give it a try if you want to.

Reply:
Do you mean using references  `std::vector<const Variable<double>&> `?

Or the variable as it is  `std::vector<const Variable<double>> `?

Using references seems to not work.

Reply:
I mean the second option. References would be the simplest, but they do not play nice with std::vector.

Reply:
Anyways, I would not hold the PR for this. Give it a try if you'd like, but otherwise we go ahead with whatever works.

Reply:
Not long ago I did experience some issues with the `std::vector<const Variable<double>>` so I'd go for the pointer option. However, I'm not so much in favour of raw pointers...